### ISSUE TITLE
Support @include for scss code

### ISSUE THREAD

**Issue by boccob** [Siilwyn/css-declaration-sorter#45]
At the moment the plugin can change a position of `@include` rule for scss code which is not safe and can lead to a wrong result.
Example:
Input
 ```scss 
a {
  @include bla;

  color: red;
  display: block;
}
```

Output
```scss
a {
  display: block;

  color: red;
  @include bla;
}
```

It'd be really good to preserve the position of `@include` rule.

**Comment by Siilwyn** [comment]
Good catch!

PostCSS scss syntax marks the `@include` as an `atrule`: https://runkit.com/embed/b31un8rzglsv

Then since it is a different type from the declarations it gets sorted to the bottom at the moment:
https://github.com/Siilwyn/css-declaration-sorter/blob/a1793968fd4b89274b237e4af036f3242c26bd65/src/index.js#L127-L129

Changing this behavior will fix it, I currently don't have access to my dev machine but will work on this soon.

**Comment by Siilwyn** [comment]
This is a bit more complex than I expected, the at-rule has to be somehow attached to the other node positions.

Came up with two solutions, which makes most sense to you @boccob?

Input:
 ```scss 
a {
  display: none;
  color: red;
  @include bla;
  border: 0;
}
```

1. Attaching to node after the at-rule:
```scss
a {
  @include bla;
  border: 0;
  color: red;
  display: none;
}
```

2. Keep it between the surrounding rules, blocking ordering:
```scss
a {
  color: red;
  display: none;
  @include bla;
  border: 0;
}
```

Think the second solution is safer with the downside of not being able to order all declarations. Will also test the performance impact on the sorting since this feature is only useful for SCSS I might put it behind a flag.

**Comment by boccob** [comment]
The 2nd one is better. Just because it can't affect result css.
Btw can I help you somehow with this?

**Comment by Siilwyn** [comment]
Thanks for your reply. Already got some patches locally, I'll let you know if things go south but otherwise I expect this to be fixed relatively easy.

**Comment by Siilwyn** [comment]
@boccob turned out to be a really small change, could you test it out before it goes into master?

To install the patched version: `npm i siilwyn/css-declaration-sorter#retain-at-rules-position`

**Comment by boccob** [comment]
Works great. Thank you!

**Comment by Siilwyn** [comment]
All right, thank you for testing! I have published the change under under version 3.0.1.
